### ISSUE TITLE
MySqlBulkCopy: MySqlConnector.MySqlException: 'Error during LOAD DATA LOCAL INFILE'

### ISSUE THREAD

**Issue by RobertGlynWilliams** [mysql-net/MySqlConnector#974]
Hi, 

Thank you for this great library! ;)

I periodically get exceptions from the MySqlBulkCopy class.
This happens when the utf8 encoder tries to encode a unicode character into a buffer with a length of 1.

Exception:
The output byte buffer is too small to contain the encoded data, encoding 'Unicode (UTF-8)' fallback 'System.Text.EncoderReplacementFallback'. (Parameter 'bytes')

Example code to reproduce:
```csharp
static async Task Main(string[] args)
{
    /*
    CREATE TABLE `testtbl` (
	    `a` MEDIUMTEXT NOT NULL,
	    `b` VARCHAR(255) NOT NULL
    )
     */

    var a = new StringBuilder();
    for (int i = 0; i < 1048573; i++)
    {
        a.Append("a");
    }

    var b = "Å string that crashes application";

    var dataTable = new DataTable();
    dataTable.Columns.Add("a");
    dataTable.Columns.Add("b");
    dataTable.Rows.Add(new object[] { a, b });

    using var connection = new MySqlConnection("...;AllowLoadLocalInfile=true");
    await connection.OpenAsync();
    var bulkCopy = new MySqlBulkCopy(connection);
    bulkCopy.DestinationTableName = "testtbl";
    await bulkCopy.WriteToServerAsync(dataTable);
}
```

**Comment by bgrainger** [comment]
Thanks for the great repro!

**Comment by bgrainger** [comment]
I naively assumed that `Convert` would output the leading bytes and maintain internal state to output the rest of the UTF-8 bytes in the next block. (And apparently I wasn't the only one: https://github.com/npgsql/npgsql/issues/2849.)

The fix will be to call `GetByteCount` to ensure that at least one `char` can be converted.

**Comment by bgrainger** [comment]
Even worse, I already made this mistake but didn't learn from it ☹ https://github.com/mysql-net/MySqlConnector/issues/520.

**Comment by RobertGlynWilliams** [comment]
Thank you for the bugfix 💯 .

**Comment by bgrainger** [comment]
Fixed in [1.3.5](https://github.com/mysql-net/MySqlConnector/releases/tag/1.3.5).

**Comment by chrisdcmoore** [comment]
Hey @bgrainger - I noticed this issue referenced an issue I opened and fixed in Npgsql a while back, and I just wanted to give you a heads up that I've discovered the fix I wrote doesn't appear to be entirely correct - I've opened another issue against Npgsql describing what I think is going on over there, so you may wish to check whether or not it also affects your fix to this issue here!

https://github.com/npgsql/npgsql/issues/3733

Cheers

**Comment by bgrainger** [comment]
Thanks @chrisdcmoore; it looks like my fix is subject to this same problem. I appreciate your bringing it to my attention.

**Comment by bgrainger** [comment]
Improved fix in [1.3.9](https://github.com/mysql-net/MySqlConnector/releases/tag/1.3.9).
